### ISSUE TITLE
Memcache persistance bug with symfony/cache: <= 3.4 (it is fixed in >= 4.0)

### ISSUE THREAD

**Issue by duronrulez** [symfony/symfony#28186]
**Symfony version(s) affected**: <= 3.4

**Description**  
Memcache persistance is not working on symfony/cache component on versions prior to 4.0

**How to reproduce**  
Provide a persistent_id to the memcache client instance.
```
$serversDsn = [['memcached://localhost:11211']];
$options = ['persistent_id' => 'test', 'namespace' => 'test'];
$adapter = MemcachedAdapter::createConnection($serversDsn, $options);
```

**Possible Solution**  
The bug is fixed in symfony/cache >= 4.0
Excerpt of fix:
location: MemcachedTrait.php
method: createConnection
code fix - taken from https://github.com/symfony/cache/blob/4.0/Traits/MemcachedTrait.php#L154
```
// set client's servers, taking care of persistent connections
            if (!$client->isPristine()) {
                $oldServers = array();
                foreach ($client->getServerList() as $server) {
                    $oldServers[] = array($server['host'], $server['port']);
                }
                $newServers = array();
                foreach ($servers as $server) {
                    if (1 < \count($server)) {
                        $server = array_values($server);
                        unset($server[2]);
                        $server[1] = (int) $server[1];
                    }
                    $newServers[] = $server;
                }
                if ($oldServers !== $newServers) {
                    $client->resetServerList();
                    $client->addServers($servers);
                }
            } else {
                $client->addServers($servers);
            }
``` 

What is the procedure (if any) to get a patch from 4.0 (or latest maintained release) in 3.4?

**Comment by nicolas-grekas** [comment]
Fixing this requires submitting a PR against branch 3.4 (which is the maintained one that has the bug apparently.)

**Comment by duronrulez** [comment]
It seems i've been using an older version of the component, the bug is fixed in v3.4.12

Closing the issue.
